src/Illuminate/Foundation/Http/Controllers/KeyGenerationController.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Illuminate\Foundation\Http\Controllers;
+
+use Illuminate\Encryption\Encrypter;
+use Illuminate\Foundation\Application;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Routing\Controller;
+use Throwable;
+
+class KeyGenerationController extends Controller
+{
+    /**
+     * The application implementation.
+     *
+     * @var \Illuminate\Foundation\Application
+     */
+    protected $app;
+
+    /**
+     * Create a new controller instance.
+     *
+     * @param  \Illuminate\Foundation\Application  $app
+     * @return void
+     */
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+    }
```

Comment:
If you want, you can use constructor property promotion

```suggestion
    /**
     * Create a new controller instance.
     *
     * @param  \Illuminate\Foundation\Application  $app
     * @return void
     */
    public function __construct(protected Application $app)
    {
    }
```